src/pages/Detail/Detail.js

```diff
@@ -1,5 +1,502 @@
+import 'react-datepicker/dist/react-datepicker.css';
+import theme from '../../styles/theme';
+
 const Detail = () => {
-  return <>상세</>;
+  const params = useParams();
+  const productId = params.id;
+  const [product, setProduct] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [isNavVisible, setIsNavVisible] = useState(false);
+  const [startDate, setStartDate] = useState(new Date());
+  const [endDate, setEndDate] = useState(new Date());
+  useEffect(() => {
+    window.addEventListener('scroll', handleScroll);
+    return () => {
+      window.removeEventListener('scroll', handleScroll);
+    };
+  }, []);
+
+  const handleScroll = () => {
+    if (window.scrollY >= 402) {
+      setIsNavVisible(true);
+    } else {
+      setIsNavVisible(false);
+    }
+  };
+
+  const changeScrollLocation = coordinate => {
+    window.scrollTo({ top: coordinate, behavior: 'smooth' });
+  };
+
+  function getDateRange(startDate, endDate, listDate) {
+    let dateMove = new Date(startDate);
+    let strDate = startDate;
+    if (startDate == endDate) {
+      strDate = dateMove.toISOString().slice(0, 10);
+      listDate.push(strDate);
+    } else {
+      while (strDate < endDate) {
+        strDate = dateMove.toISOString().slice(0, 10);
+        listDate.push(strDate);
+        dateMove.setDate(dateMove.getDate() + 1);
+      }
+    }
+    return listDate;
+  }
+
+  useEffect(() => {
+    fetch(`http://10.58.52.51:3000/product/${productId}`, {
+      method: 'GET',
+    })
+      .then(response => response.json())
+      .then(result => {
+        setProduct(result);
+        setLoading(false);
+      });
+  }, [productId]);
+
+  const makeInputToPriceFormat = number => {
+    let num = Number.parseFloat(number).toFixed(0);
+    const comma = num => {
+      num = String(num);
+      return num.replace(/(\d)(?=(?:\d{3})+(?!\d))/g, '$1,');
+    };
+    const uncomma = num => {
+      num = String(num);
+      return num.replace(/[^\d]+/g, '');
+    };
+    return comma(uncomma(num));
+  };
+  if (loading) return;
+
+  if (!product.data[0].reservations) {
+    product.data[0].reservations = [
+      {
+        check_in: '1111-11-11',
+        check_out: '1111-11-12',
+      },
+    ];
+  } else {
+  }
+  if (!product.data[0].holidays) {
+    product.data[0].holidays = [
+      {
+        check_in: '1111-11-13',
+        check_out: '1111-11-14',
+      },
+    ];
+  } else {
+  }
+  const pictureArrayWithoutThumbnail = product.data[0].imageUrl.slice(1);
+
+  let excludedDateArray = [];
+  const reservedAndHolidaysArray = [
+    ...product.data[0].reservations,
+    ...product.data[0].holidays,
+  ];
+
+  {
+    reservedAndHolidaysArray.map(el => {
+      return getDateRange(el.check_in, el.check_out, excludedDateArray);
+    });
+  }
+
+  excludedDateArray = [...new Set(excludedDateArray)];
+  excludedDateArray.sort(function (a, b) {
+    return a - b;
+  });
```

Comment:
이건 어때요? 명령형으로 되어 있는 코드를 선언형으로 수정 하면 좋을 것 같습니다.
```suggestion
이부분은 같이 확인해보면서 수정해보시죠!
```